charts/aws-vpc-cni/crds/customresourcedefinition.yaml

```diff
@@ -14,6 +14,41 @@ spec:
         openAPIV3Schema:
           type: object
           x-kubernetes-preserve-unknown-fields: true
+          description: ENIConfig is the Schema for the eniconfigs API
+          properties:
+            apiVersion:
+              description: |-
+                APIVersion defines the versioned schema of this representation of an object.
+                Servers should convert recognized schemas to the latest internal value, and
+                may reject unrecognized values.
+                More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
+              type: string
+            kind:
+              description: |-
+                Kind is a string value representing the REST resource this object represents.
+                Servers may infer this from the endpoint the client submits requests to.
+                Cannot be updated.
+                In CamelCase.
+                More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+              type: string
+            metadata:
+              type: object
+            spec:
+              description: ENIConfigSpec defines the desired state of ENIConfig
+              properties:
+                securityGroups:
+                  items:
+                    type: string
+                  type: array
+                subnet:
+                  type: string
+              required:
+              - securityGroups
```

Comment:
Security groups aren't required

Reply:
This was auto generated from this - https://github.com/aws/amazon-vpc-cni-k8s/blob/189f00f64f3e0b0bd0f3b8eff070aed3f0ba32b6/pkg/apis/crd/v1alpha1/eniconfig_types.go#L28

Since both securityGroup and subnets are listed in spec, it is taking them as required.

Reply:
Once this is enforced ENIConfigs can't be applied unless securityGroups are specified

Reply:
Yeah, we discussed internally about this since it would be needed for upgrades and downgrades..